pulsar-io/alluxio/src/main/java/org/apache/pulsar/io/alluxio/sink/AlluxioAbstractSink.java

```diff
@@ -0,0 +1,236 @@
+import org.apache.pulsar.io.core.KeyValue;
+import org.apache.pulsar.io.core.Sink;
+import org.apache.pulsar.io.core.SinkContext;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * A Simple abstract class for Alluxio sink
+ * Users need to implement extractKeyValue function to use this sink
+ */
+@Slf4j
+public abstract class AlluxioAbstractSink<K, V> implements Sink<V> {
+
+    private FileSystem fileSystem;
+    private FileOutStream fileOutStream;
+    private CreateFileOptions createFileOptions;
+    private long recordsNum;
+    private String tmpFilePath;
+    private String fileDirPath;
+    private String tmpFileDirPath;
+    private long lastRotationTime;
+    private long rotationRecordsNum;
+    private long rotationInterval;
+    private AlluxioSinkConfig alluxioSinkConfig;
+    private AlluxioState alluxioState;
+
+    private List<Record<V>> recordsToAck;
+
+    @Override
+    public void open(Map<String, Object> config, SinkContext sinkContext) throws Exception {
+        alluxioSinkConfig = AlluxioSinkConfig.load(config);
+        alluxioSinkConfig.validate();
+
+        // initialize FileSystem
+        String alluxioMasterHost = alluxioSinkConfig.getAlluxioMasterHost();
+        int alluxioMasterPort = alluxioSinkConfig.getAlluxioMasterPort();
+        Configuration.set(PropertyKey.MASTER_HOSTNAME, alluxioMasterHost);
+        Configuration.set(PropertyKey.MASTER_RPC_PORT, alluxioMasterPort);
+        if (alluxioSinkConfig.getSecurityLoginUser() != null) {
+            Configuration.set(PropertyKey.SECURITY_LOGIN_USERNAME, alluxioSinkConfig.getSecurityLoginUser());
+        }
+        fileSystem = FileSystem.Factory.get();
+
+        // initialize alluxio dirs
+        String alluxioDir = alluxioSinkConfig.getAlluxioDir();
+        fileDirPath = alluxioDir.startsWith("/") ? alluxioDir : "/" + alluxioDir;
+        tmpFileDirPath = fileDirPath + "/tmp";
+
+        AlluxioURI alluxioDirPath = new AlluxioURI(fileDirPath);
+        if (!fileSystem.exists(alluxioDirPath)) {
+            fileSystem.createDirectory(alluxioDirPath);
+        }
+
+        AlluxioURI tmpAlluxioDirPath = new AlluxioURI(tmpFileDirPath);
+        if (!fileSystem.exists(tmpAlluxioDirPath)) {
+            fileSystem.createDirectory(tmpAlluxioDirPath);
+        }
+
+        createFileOptions = CreateFileOptions.defaults();
+
+        recordsNum = 0;
+        recordsToAck = Lists.newArrayList();
+        tmpFilePath = "";
+        alluxioState = AlluxioState.WRITE_STARTED;
+
+        lastRotationTime = System.currentTimeMillis();
+        rotationRecordsNum = alluxioSinkConfig.getRotationRecords();
+        rotationInterval =  alluxioSinkConfig.getRotationInterval();
+    }
+
+    @Override
+    public void write(Record<V> record) {
+        long now = System.currentTimeMillis();
+
+        switch (alluxioState.ordinal()) {
```

Comment:
why not 

```
switch (alluxioState) {
case WRITE_STARTED
}
```

It is much easy to maintain rather than using ordinal

Reply:
Changed, thanks